### PR TITLE
Fix types error for api-controller and demoDataSetup

### PR DESCRIPTION
**Pull request by JameelB**
## Description
Fix Typescript types errors

## Additional Notes
Related Gist - https://gist.github.com/JameelB/bca467389fd1b58ea027b36b625e9793

### CODE REVIEW

File: demo/server/src/modules/index.ts [hunk 1]
```diff
@@ -94,7 +94,7 @@ function wfmApiSetup(app: express.Router, connectionPromise: Promise<any>) {
   });
 }
 
-function demoDataSetup(connectionPromise: Promise<Db>) {
+function demoDataSetup(connectionPromise: Promise<any>) {
```
Comment (paolobueno) [inline demo/server/src/modules/index.ts]: If the promise that's being passed here is really `Db | undefined` from line 31 I think we should look into where the undefined is coming from, might introduce a bug.

**Review by paolobueno — DISMISSED** [review]
Great fix, just a little bit worried about the mongodb being possibly undefined.

File: demo/server/src/modules/index.ts (hunk 1 above)
Comment (JameelB) [reply demo/server/src/modules/index.ts]: That's due to the function that's giving back the connectionPromise as that may be void. https://github.com/feedhenry-raincatcher/raincatcher-core/blob/master/demo/server/src/modules/index.ts#L66-L83

Comment (paolobueno) [reply demo/server/src/modules/index.ts]: Yes, might be worth it to handle that case in this function, even if it's just erroing out and printing it to the user.

**Review by wtrocki — APPROVED** [review]

**Pull request merged**